sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/higherOrderFunctions.scala

```diff
@@ -409,16 +424,23 @@ case class ArrayExists(
   override def nullSafeEval(inputRow: InternalRow, argumentValue: Any): Any = {
     val arr = argumentValue.asInstanceOf[ArrayData]
     val f = functionForEval
-    var exists = false
+    var foundNull = false
     var i = 0
-    while (i < arr.numElements && !exists) {
+    while (i < arr.numElements) {
       elementVar.value.set(arr.get(i, elementVar.dataType))
-      if (f.eval(inputRow).asInstanceOf[Boolean]) {
-        exists = true
+      val ret = f.eval(inputRow)
+      if (ret == null) {
+        foundNull = true
+      } else if (ret.asInstanceOf[Boolean]) {
+        return true
```

Comment:
can we avoid using `return` here and keep the previous way we handle `exists`? Using return is a pretty bad practice and I think here having an extra flag we can easily avoid it..

Reply:
+1 we don't have to use early return here. The old code works fine and conveys the loop condition well.

Reply:
Sure, I updated to use `exists` back.